### PR TITLE
Migrate Flex to typescript

### LINKED ISSUE #909: Flex Typescript Migration
TypeScript here we come!
We are migrating to typescript, we are doing so component by component

We would love some help in converting some of our component to typescript - we've created a README
https://github.com/mondaycom/monday-ui-react-core/blob/master/TYPESCRIPT_MIGRATION.md

in this PR we expect you to convert the following files

- [ ] Flex.jsx
- [ ] FlexConstants.js
- [ ] flex-snapshot-tests.jest.js

Good Luck

### PR DESCRIPTION
**Pull request by piotrski**
This PR migrates `Flex` component to typescript.
Resolves #909 

#### Basic
- [ ] Used plop (`npm run plop`) to create a new component.
- [ ] PR has description.
- [ ] New component is functional and uses Hooks. 
- [ ] Component defines [`PropTypes`](https://reactjs.org/docs/typechecking-with-proptypes.html).
#### Style
- [ ] Styles are added to `NewComponent.modules.scss` file inside of the `NewComponent` folder.
- [ ] Component uses CSS Modules.
- [x] Design is compatible with [Monday Design System](https://design.monday.com/).
#### Storybook
- [x] Stories were added to `/src/NewComponent/__stories__/NewComponent.stories.js` file.
- [x] Stories include all flows of using the component.
#### Tests
- [x] Tests are compliant with [TESTING_README.md](TESTING_README.md) instructions.

### CODE REVIEW

File: src/components/Flex/Flex.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  elementType?: React.ElementType;
+  wrap?: boolean;
+  children?: React.ReactElement;
+  justify?: FLEX_JUSTIFY;
+  align?: FLEX_ALIGN;
+  gap?: FLEX_GAPS | number;
+  ariaLabel?: string;
+  tabIndex?: number;
+  /** onClick function - MouseEvent */
+  onClick?: React.MouseEventHandler;
```
Comment (orrgottlieb) [inline src/components/Flex/Flex.tsx:26]: Hi @piotrski
the onClick could also be from a keyboard event, please add it as well

File: src/components/Flex/Flex.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  children: undefined,
+  direction: FLEX_DIRECTIONS.ROW,
+  justify: FLEX_JUSTIFY.START,
+  align: FLEX_ALIGN.CENTER,
+  gap: FLEX_GAPS.NONE,
+  ariaLabel: undefined,
+  tabIndex: undefined
+};
+
+export default {
```
Comment (orrgottlieb) [inline src/components/Flex/Flex.tsx:101]: for consistency in the code base, 
please export default the Component and assign `justify`, `align`, `gaps` and `directions` as properties to the component 🙏

File: src/components/Flex/__stories__/Flex.stories.mdx [hunk 3]
```diff
@@ -6,10 +6,19 @@ import classes from "./Flex.stories.module.scss";
 import Chips from "../../Chips/Chips";
 import { StoryDescription } from "../../../storybook/components/story-description/story-description";
 import { LIST, MENU, TABS } from "../../../storybook/components/related-components/component-description-map";
+import { createStoryMetaSettings } from "../../../storybook/functions/create-component-story";
+
+export const metaSettings = createStoryMetaSettings({
```
Comment (orrgottlieb) [inline src/components/Flex/__stories__/Flex.stories.mdx:11]: this is awesome, thank you 🙏

File: src/components/Flex/Flex.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  gap?: FLEX_GAPS | number;
+  ariaLabel?: string;
+  tabIndex?: number;
+  /** onClick function - MouseEvent */
+  onClick?: React.MouseEventHandler;
+  /** element id to describe the counter accordingly */
+  ariaLabelledby?: string;
+};
+
+const Flex: VibeComponent<FlexProps, HTMLElement> = forwardRef(
```
Comment (orrgottlieb) [inline src/components/Flex/Flex.tsx:31]: please keep the `propTypes` on the Component itself 🙏

**Review by orrgottlieb — CHANGES_REQUESTED** [review]
This is an awesome job!
i have some small comments 🙏 
thank you so much

**Comment by SergeyRoyt** [comment]
Closed in favor of https://github.com/mondaycom/monday-ui-react-core/pull/1006

**Pull request closed without merging**